openbook_auth/models.py

```diff
@@ -2576,6 +2589,17 @@ def _check_follow_list_id(self, list_id):
 
     def _check_can_update_post_with_id(self, post_id):
         self._check_has_post_with_id(post_id=post_id)
+        Post = get_post_model()
+        post = Post.objects.get(id=post_id)
+        is_community_post = Post.is_post_with_id_a_community_post(post_id)
+        if post.is_closed and is_community_post:
+            is_administrator = self.is_administrator_of_community_with_name(post.community.name)
```

Comment:
I added a method `is_staff_of_community_with_name` which checks whether its admin or mod of community.